pybamm_tea/tea.py

```diff
@@ -0,0 +1,690 @@
+#
+# TEA class
+#
+
+import pybamm
+import matplotlib.pyplot as plt
+import matplotlib.patches as patches
+import pandas as pd
+
+
+class TEA:
+    """
+    A Techno-Economic Analysis class for estimation of cell metrics.
+
+    Parameters
+    ----------
+    parameter_values : :class:pybamm.ParameterValues, optional
+        A dictionary of parameters and their corresponding numerical values.
+    inputs : dict
+        A dictionary of inputs (additional parameters) and their corresponding
+        numerical values.
+    """
+
+    def __init__(self, parameter_values=None, inputs=None):
+        self.parameter_values = parameter_values or pybamm.ParameterValues({})
+        self.inputs = inputs or {}
+
+        # Update parameter values with inputs
+        self.parameter_values.update(self.inputs, check_already_exists=False)
+        pava = self.parameter_values
+
+        # Raise error if electrolyte density is missing
+        if pava.get("Electrolyte density [kg.m-3]") is None:
+            raise ValueError(
+                "The parameter 'Electrolyte density [kg.m-3]' must be provided"
+            )
+
+        # Update inactive material density, active material volume fraction, dry
```

Comment:
these calculations could be done in a separate method that gets called in the init